base/strings/search.jl

```diff
@@ -166,35 +188,40 @@ function _search_bloom_mask(c)
 end
 
 _nthbyte(s::String, i) = codeunit(s, i)
-_nthbyte(a::Union{AbstractVector{UInt8},AbstractVector{Int8}}, i) = a[i]
+_nthbyte(a::Union{Vector{UInt8},Vector{Int8}}, i) = a[i]
```

Comment:
You can eliminate this definition — it is redundant with the one below based on `firstindex(t)`, and LLVM is smart enough to eliminate the extra additions when `firstindex(t)` is inlined to `1` at compile-time.